### PR TITLE
Doc tests

### PR DESCRIPTION
**Pull request by bendichter**
## Motivation

Refactor docs for more extensive documentation of testing

## Checklist

- [ ] Did you update CHANGELOG.md with your changes?
- [ ] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [ ] Have you ensured the PR clearly describes the problem and the solution?
- [ ] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [ ] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [ ] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

File: docs/source/testing/index.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Testing
+-------
+
+PyNWB has a goal of 100% test coverage, so it is important that any changes to the code base be covered by tests. Our
+tests are split into two four categories, all of which are in the ``tests/`` folder:
```
Comment (oruebel) [inline docs/source/testing/index.rst]:
```suggestion
tests are split into three main categories, all of which are in the ``tests/`` folder:
```

File: docs/source/testing/index.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+* ``tests/back_compat`` is a collection of small old NWB files and a small testing script that ensures we can still
+  read them.
```
Comment (oruebel) [inline docs/source/testing/index.rst]:
```suggestion
* ``tests/back_compat`` tests to check compatibility of the API with older version of NWB. This is a collection of small NWB files with many different versions and a small testing script that ensures we can still read them.
```

File: docs/source/testing/index.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+* ``tests/unit``: tests that cover each use-case of each method or function
+* ``tests/integration``: tests that neurodata types and NWB file can be written and read in various modes and
+  backends including HDF5 and using the ros3 driver.
+
```
Comment (oruebel) [inline docs/source/testing/index.rst]: @rly @CodyCBakerPhD what about https://github.com/NeurodataWithoutBorders/pynwb/tree/dev/tests/validation it looks like that folder is intended for tests of command line tools. Should this be renamed to tests/cli_tools ?

File: docs/source/testing/index.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+In addition to the ``tests/`` folder, there is also a :py:mod:`pynwb.testing` package. This module does not
+contain any tests but instead contains classes and/or functions that can be used by PyNWB tests and can be imported
+and used in the testing suite of any downstream library.
```
Comment (oruebel) [inline docs/source/testing/index.rst]:
```suggestion
contain any tests but instead contains classes and/or functions that help with the implementation of tests for PyNWB and are used in the testing suite and also be imported
and used by downstream libraries, e.g., to implement test suites for extensions to the NWB (NDX).
```

File: docs/source/testing/index.rst (hunk 3 above)
Comment (bendichter) [reply docs/source/testing/index.rst]: I added a mention of this in the docs

File: docs/source/testing/index.rst [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+-------
+
+PyNWB has a goal of 100% test coverage, so it is important that any changes to the code base be covered by tests. Our
+tests are split into four main categories, all of which are in the ``tests/`` folder:
+
+* ``tests/back_compat`` tests to check compatibility of the API with older version of NWB. This is a collection of small NWB files with many different versions and a small testing script that ensures we can still read them.
+* ``tests/unit``: tests that cover each use-case of each method or function
+* ``tests/integration``: tests that neurodata types and NWB file can be written and read in various modes and
+  backends including HDF5 and using the ros3 driver.
+* ``tests/validation``: tests command line usage of NWB validator.
```
Comment (oruebel) [inline docs/source/testing/index.rst]:
```suggestion
* ``tests/validation``: tests command line usage of the ``pynwb.validate`` NWB validator.
```

**Review by oruebel — COMMENTED** [review]
Can you update the Changelog. Otherwise this looks good to me. Thanks for adding this documentation!

**Review by oruebel — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**